### DISCUSSION TITLE
How can I get "_U" cookie from bing.com?

### DISCUSSION THREAD

**Discussion by taishi55** [waylaidwanderer/node-chatgpt-api#58] [Help] [answered]
Hi, sorry if this is a simple question. 

I have a JS code to get "_U" cookie from a page. But I don't know how to use it on bing.com.

The blow is my funciton.
```ts
function getCookie() {
  const cookieName = "_U";
  const cookies = document.cookie.split(';');
  for (let i = 0; i < cookies.length; i++) {
    const cookie = cookies[i].trim();
    if (cookie.startsWith(cookieName + '=')) {
      return decodeURIComponent(cookie.substring(cookieName.length + 1));
    }
  }
  return null;
}
```

**Comment by waylaidwanderer** [comment, marked as answer]
You don't use it on bing.com; you go to bing.com and get the `_U` cookie (you can grab it from DevTools), and then you give it to `BingAIClient` to use it.

**Comment by taishi55** [comment]
Thank you!! By chance, is there any way to grab the key using iframe?

**Comment by waylaidwanderer** [comment]
I think you can just grab it manually. It lasts indefinitely so you don't need to worry about refreshing it.
